src/org/sparkboard/server/server.clj

```diff
@@ -95,18 +104,19 @@
                                {:view_id view-id
                                 :view (hiccup/->blocks-json (screens/team-broadcast-modal-compose context))}))
 
-      "user:team-broadcast-response"
-      (slack/web-api "views.open" {:auth/token (:slack/bot-token context)}
-                     {:trigger_id (:trigger_id payload)
-                      :view (hiccup/->blocks-json
-                             (screens/team-broadcast-response
-                              (->> payload :message :blocks first :text :text) ; broadcast msg
-                              (->> payload :message :blocks last
-                                   :elements first :text
-                                   ;; text between brackets with lookahead/lookbehind:
-                                   (re-find #"(?<=\[).+?(?=\])"))))})
-
-      "broadcast2:channel-select"                           ;; refresh same view then save selection in private metadata
+      "user:team-broadcast-response" ; "Post an Update" button (user opens modal to respond to broadcast)
+      (let [firebase-path (str "/slack-broadcast/" firebase-key)
+            firebase-child-ref (.push (fire-jvm/->ref (str firebase-path "/replies")))]
+        (fire-jvm/set-value firebase-child-ref
+                            {:responding-from-channel (-> payload :channel :name)})
+        (slack/web-api "views.open" {:auth/token (:slack/bot-token context)}
+                       {:trigger_id (:trigger_id payload)
+                        :view (hiccup/->blocks-json
+                               (screens/team-broadcast-response
+                                (->  payload :message :blocks first :text :text) ; broadcast msg
```

Comment:
I was surprised to see they don’t give us a nice :state map here. Looking things up positionally feels so brittle, maybe we should write a fn that indexes all the blocks by id so we can safely rearrange views without breaking things like this. (Not necessary for this PR)